generator/lib/createPointCloudTile.js

```diff
@@ -0,0 +1,554 @@
+
+    featureTableJson.POINTS_LENGTH = pointsLength;
+
+    if (defined(constantColor)) {
+        featureTableJson.CONSTANT_RGBA = constantColor;
+    }
+
+    if (quantizePositions) {
+        // Quantized offset is the lower left, unlike RTC_CENTER which is the center
+        featureTableJson.QUANTIZED_VOLUME_SCALE = [tileWidth, tileWidth, tileWidth];
+        featureTableJson.QUANTIZED_VOLUME_OFFSET = [center.x - radius, center.y - radius, center.z - radius];
+    } else if (relativeToCenter){
+        featureTableJson.RTC_CENTER = [center.x, center.y, center.z];
+    }
+
+    if (batched) {
+        featureTableJson.BATCH_LENGTH = batchIds.batchLength;
+    }
+
+    for (i = 0; i < attributesLength; ++i) {
+        attribute = attributes[i];
+        featureTableJson[attribute.propertyName] = {
+            byteOffset : attribute.byteOffset,
+            componentType : attribute.componentType // Only defined for batchIds
+        };
+        attribute.buffer.copy(featureTableBinary, attribute.byteOffset);
+    }
+
+    var batchTable;
+    var batchTableJson;
+    var batchTableBinary;
+
+    if (batched) {
+        batchTable = getBatchTableForBatchedPoints(batchIds.batchLength);
+        batchTableJson = batchTable.json;
+        batchTableBinary = batchTable.binary;
+    } else if (perPointProperties) {
+        batchTable = getBatchTableForPerPointProperties(pointsLength, noiseValues);
+        batchTableJson = batchTable.json;
+        batchTableBinary = batchTable.binary;
+    }
+
+    var pnts = createPnts({
+        featureTableJson : featureTableJson,
+        featureTableBinary : featureTableBinary,
+        batchTableJson : batchTableJson,
+        batchTableBinary : batchTableBinary
+    });
+
+    return {
+        pnts : pnts,
+        batchTableJson : batchTableJson
+    };
+}
+
+// Return a position in the range of (-0.5, -0.5, -0.5) to (0.5, 0.5, 0.5) based on the index
+function getPosition(i, pointsLength) {
+    var width = Math.round(Math.pow(pointsLength, 1/3));
+    var z = Math.floor(i / (width * width));
+    var y = Math.floor((i - z * width * width) / width);
+    var x = i - width * (y + width * z);
+
+    x = x / width - 0.5;
+    y = y / width - 0.5;
+    z = z / width - 0.5;
+
+    return new Cartesian3(x, y, z);
+}
+
+function boxFunction(i, pointsLength, radius) {
+    var position = getPosition(i, pointsLength);
+    Cartesian3.multiplyByScalar(position, radius, position);
+    return position;
+}
+
+function sphereFunction(i, pointsLength, radius) {
+    void(i, pointsLength); // Unused parameters
+    var theta = CesiumMath.nextRandomNumber() * 2 * Math.PI;
+    var phi = CesiumMath.nextRandomNumber() * Math.PI - Math.PI/2.0;
+    var x = radius * Math.cos(theta) * Math.cos(phi);
+    var y = radius * Math.sin(phi);
+    var z = radius * Math.sin(theta) * Math.cos(phi);
+    return new Cartesian3(x, y, z);
+}
+
+function randomFunction(position) {
+    void(position); // Unused parameter
```

Comment:
Is this to pass JSHint?  There might be a JSHint comment to disable this warning in this function like we do for other things.